src/cocoa/toga_cocoa/widgets/internal/cells.py

```diff
@@ -24,31 +28,39 @@
     NSLayoutConstraint,
     NSLayoutRelationEqual,
     NSLineBreakMode,
+    NSButton,
+    NSSwitchButton,
     at,
     objc_method,
 )
 
 
-class TogaIconView(NSTableCellView):
+class TogaTableCellView(NSTableCellView):
+
+    imageView = objc_property(objc_id)
+    checkbox = objc_property(objc_id)
+    textField = objc_property(objc_id)
 
     @objc_method
-    def setup(self):
-        image_view = NSImageView.alloc().init()
-        text_field = NSTextField.alloc().init()
+    def initWithLayout(self):
+        send_super(__class__, self, "init")
+        self.imageView = NSImageView.alloc().init()
+        self.checkbox = NSButton.alloc().init()
+        self.textField = NSTextField.alloc().init()
```

Comment:
That's a good question. In my experience, the overhead has been small since in practice only the number of TogaTableCellView instances that are actually visible on screen will be created (maybe a few more than that) and then reused when scrolling.

We might gain a bit by lazily allocating images and checkboxes as needed while slightly complicating the logic.